### PR TITLE
Update ingress resources

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR does not touch the actual code of the web UI. 

The changes are in the helm chart:

- Change certificate secret type to kubernetes/tls. This allows cert-exporter to monitor a possible expiry
- Update the Ingress resources to networking.k8s.io/v1. This prevents installing on kubernetes below 1.19
- Update ClusteRole and ClusterRoleBinding apiVersion
- Use `ingress.tls` values for certificate secrets. This is the same as used by athena and dex-app
- Put happaapi Ingress and secret into `giantswarm` namespace instead of `default`

### CODE REVIEW

**Review by kuosandys — APPROVED** [review]
👍

File: helm/happa/Chart.yaml [hunk 1]
```diff
@@ -7,3 +7,4 @@ version: [[ .Version ]]
 appVersion: [[ .AppVersion ]]
 annotations:
   config.giantswarm.io/version: 1.x.x
+kubeVersion: ">=1.19.0-0"
```
Comment (marians) [inline helm/happa/Chart.yaml:10]: ❤️

File: helm/happa/templates/happaapi-ingress.yaml [hunk 2]
```diff
@@ -1,12 +1,8 @@
-{{ if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
 apiVersion: networking.k8s.io/v1
-{{ else }}
-apiVersion: networking.k8s.io/v1beta1
-{{ end }}
 kind: Ingress
 metadata:
   name: happaapi
-  namespace: default
+  namespace: giantswarm
```
Comment (marians) [inline helm/happa/templates/happaapi-ingress.yaml:5]: Have you tested the namespace change thoroughly?

This ingress is a proxy to the Kubernetes API server (see `...backend.service.name: kubernetes`). I have tried moving the ingress resource to the giantswarm namespace in the past, and as a result, happaapi.* didn't work any more.

Comment (ubergesundheit) [reply helm/happa/templates/happaapi-ingress.yaml:5]: No I didn't test this, but good point. I'll see what I can find!

Comment (ubergesundheit) [reply helm/happa/templates/happaapi-ingress.yaml]: You're right! Reverted my changes.

**Pull request merged**